packages/x-data-grid-pro/src/tests/dataSource.DataGridPro.test.tsx

```diff
@@ -40,8 +41,15 @@ describeSkipIf(isJSDOM)('<DataGridPro /> - Data source', () => {
       { useCursorPagination: false, minDelay: 0, maxDelay: 0, verbose: false },
       shouldRequestsFail,
     );
-    fetchRowsSpy = spy(mockServer, 'fetchRows');
-    const { fetchRows } = mockServer;
+
+    // Reuse the same stub between rerenders to properly count the calls
+    fetchRowsSpy = useLazyRef(() => stub()).current;
```

Comment:
The reason `setProps` was causing issues is that `fetchRowsSpy` was recreated on each `TestDataSource` rerender 😅
I've fixed that and reverted back the usage of `setProps`.
Some tests will probably fail though, but IMO in that case there's something wrong with the test or implementation.